File: EECS4413Project/src/main/java/model/Catalog.java
package model;

import java.util.ArrayList;

import bean.ItemBean;

public class Catalog {
	ArrayList<ItemBean> items = new ArrayList<>();
	
	public Catalog(){}
	public void reset() {
		items = new ArrayList<>();
	}
	public void addItem(ItemBean item) {
		items.add(item);
	}
	
	public void removeItem(ItemBean item) {
		items.remove(item);
	}
	
	public Catalog filter(Boolean apple, Boolean samsung, Boolean phone, Boolean laptop) {
		Catalog cat = new Catalog();
		for(ItemBean item: items) {
			if(apple && item.getBrand().equals("apple")) 
				cat.addItem(item);
			else if(samsung && item.getBrand().equals("samsung")) 
				cat.addItem(item);
			else if(phone && item.getType().equals("phone")) 
				cat.addItem(item);
			else if(laptop && item.getType().equals("laptop"))
				cat.addItem(item);
		}
		return cat;
	}
}

File: EECS4413Project/src/main/webapp/scripts/catalog.js
function makeCatalogRequest(url){
	var request= new XMLHttpRequest(); //prepare an HTTP request
	request.open("GET", (url), true);
	request.onreadystatechange = function() {
		handler(request);
	};
	request.send();
}


function handler(request){
  if ((request.readyState == 4) && (request.status == 200)){
	 var target = document.getElementById("cat");
     var rs=JSON.parse(request.responseText);
     makeCatalog(rs, target)
   }
} 

function makeCatalog(rs, target){     
	//reset catalog 
    var child = target.lastElementChild; 
    while (child) {
        target.removeChild(child);
        child = target.lastElementChild;
    }

	for(var i = 0; i < rs.items.length; i++){
		var item = rs.items[i];
		var div = document.createElement('div');
		div.className="col-12 col-md-4 col-lg-3 mb-5";
		var a = document.createElement('a');
		a.className="product-item";
		a.href="product.html?name="+item.name+"&description="+item.description+"&bid="+item.bid+"&img="+item.img+"&price="+item.price;
		var img = document.createElement('img');
		img.src = item.img;
		img.className="img-fluid product-thumbnail";
		img.width=200;
		img.height=200;
		a.appendChild(img);
		var header = document.createElement('h3');
		header.innerText = item.name;
		header.className="product-title"
		a.appendChild(header);
		var price = document.createElement('strong');
		price.innerText = '$'+ item.price;
		price.className="product-price"
		a.appendChild(price);
		div.appendChild(a);
		target.appendChild(div);
	}
}

function filterCatalog(url){
	var request= new XMLHttpRequest(); //prepare an HTTP request
	var data = '';
	if(document.getElementById("brand1").checked) data+= "apple=true";
	else data+="apple=false";
	
	if(document.getElementById("brand2").checked) data+= "&samsung=true";
	else data+="&samsung=false";
	
	if(document.getElementById("type1").checked) data+= "&phone=true";
	else data+="&phone=false";
	
	if(document.getElementById("type2").checked) data+= "&laptop=true";
	else data+="&laptop=false";
	
	request.open("GET", (url + "?" + data), true);
	request.onreadystatechange = function() {
		handler(request);
	};
	request.send();
}